### PR TITLE
Fixed configs and added new configs for filters performance benchmarks.

### PR DESCRIPTION
**Pull request by navneet1v**
### Description
Fixed configs and added new configs for filters performance benchmarks.
 
### Issues Resolved
NA
 
### Check List
- [X] New functionality includes testing.
  - [X] All tests pass
- [X] New functionality has been documented.
  - [X] New functionality has javadoc added
- [X] Commits are signed as per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/k-NN/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: benchmarks/perf-tool/release-configs/faiss-hnsw/filtering/restrictive-filter/restrictive-filter-test.yml [hunk 1]
```diff
@@ -8,13 +9,13 @@ steps:
     index_name: target_index
   - name: create_index
     index_name: target_index
-    index_spec: [INDEX_SPEC_PATH]/index.json
+    index_spec: /home/ec2-user/k-NN/benchmarks/perf-tool/release-configs/faiss-hnsw/filtering/restrictive-filter/index.json
```
Comment (heemin32) [inline benchmarks/perf-tool/release-configs/faiss-hnsw/filtering/restrictive-filter/restrictive-filter-test.yml:12]: Intentional change?

File: benchmarks/perf-tool/release-configs/faiss-hnsw/filtering/relaxed-filter/relaxed-filter-test.yml [hunk 2]
```diff
@@ -8,17 +9,23 @@ steps:
...
     field_name: target_field
     bulk_size: 500
     dataset_format: hdf5
-    dataset_path: [DATASET_PATH]/sift-128-euclidean-with-attr.hdf5
+    dataset_path: /home/ec2-user/k-NN/benchmarks/perf-tool/dataset/sift-128-euclidean-with-attr.hdf5
     attributes_dataset_name: attributes
     attribute_spec: [ { name: 'color', type: 'str' }, { name: 'taste', type: 'str' }, { name: 'age', type: 'int' } ]
   - name: refresh_index
     index_name: target_index
+<<<<<<< HEAD
```
[the remark this answers is not in the document]
Comment (navneet1v) [reply benchmarks/perf-tool/release-configs/faiss-hnsw/filtering/relaxed-filter/relaxed-filter-test.yml:23]: Got missed. will fix it

File: benchmarks/perf-tool/release-configs/faiss-ivf/filtering/relaxed-filter/relaxed-filter-test.yml [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+test_name: "Faiss IVF Relaxed Filter Test"
+test_id: "Faiss IVF Relaxed Filter Test"
+num_runs: 10
+show_runs: false
+setup:
+  - name: delete_index
+    index_name: train_index
+  - name: create_index
+    index_name: train_index
+    index_spec: /home/ec2-user/[PATH]/release-configs/faiss-ivf/filtering/relaxed-filter/train-index-spec.json
```
[the remark this answers is not in the document]
Comment (navneet1v) [reply benchmarks/perf-tool/release-configs/faiss-ivf/filtering/relaxed-filter/relaxed-filter-test.yml:11]: There is not a lot of symmetry in the configs. but let me try to fix it for all the places.

**Comment by navneet1v** [comment]
@heemin32 fixed all the comments

**Review by heemin32 — APPROVED** [review]

**Review by martin-gaievski — APPROVED** [review]

**Pull request merged**